### PR TITLE
Support AWS Provider V5

### PR DESCRIPTION
**Pull request by max-lobur**
## what

Support AWS Provider V5
Linter fixes

## why

Maintenance

## references

https://github.com/hashicorp/terraform-provider-aws/releases/tag/v5.0.0

### CODE REVIEW

File: examples/complete/main.tf [hunk 1]
```diff
@@ -3,26 +3,22 @@ provider "aws" {
...
+  source               = "cloudposse/dynamic-subnets/aws"
+  version              = "2.3.0"
   availability_zones   = var.availability_zones
   vpc_id               = module.vpc.vpc_id
-  igw_id               = module.vpc.igw_id
-  cidr_block           = module.vpc.vpc_cidr_block
-  nat_gateway_enabled  = true
+  igw_id               = [module.vpc.igw_id]
+  ipv4_cidr_block      = [module.vpc.vpc_cidr_block]
+  nat_gateway_enabled  = false
```
Comment (zdmytriv) [inline examples/complete/main.tf:19]: `nat_gateway_enabled=false`?

Comment (max-lobur) [reply examples/complete/main.tf:22]: It's a test, so it's fine

**Review by zdmytriv — APPROVED** [review]

**Pull request merged**